### ISSUE TITLE
[Godot 4.0] External Editor Shows Error when using class_name

### ISSUE THREAD

**Issue by TicklesTheBrain** [godotengine/godot#65255]
### Godot version

4.0 alpha 15

### System information

Windows 10, Radeon 5500 XT

### Issue description

External editor displays message "Class X hides a global script class", whenever saving a file with class_name.

I am using VSCode.

![image](https://user-images.githubusercontent.com/21967696/188194556-2b429c43-4a76-4157-9b16-aaf329854b88.png)

-Everything works fine when running the code, just the external editor error that is distracting and annoying.
-Internal editor doesn't complain
-The error doesn't appear when external editor is loaded or after clicking "reload project" in godot editor, but appears after saving the file.

### Steps to reproduce

1. Open attached project.
2. Open SimpleClass.gd with external editor.
3. Save file
4. There's an error.

### Minimal reproduction project

[godot_test.zip](https://github.com/godotengine/godot/files/9479528/godot_test.zip)

**Comment by TicklesTheBrain** [comment]
This no longer happens in Godot 4.0 Beta 1

**Comment by Alikae** [comment]
Still happened to me in Beta1.
While trying to reproduce it in the smallest possible project, I succeeded, and after removing some more lines I couldn't reproduce it again, neither in the small nor the big project.

I managed to see the same error by creating two scripts with the same classname (That was not the case with the initial bug/reproduction).

My wild guess would be that for some reason, the parser parse the new file while the old version of the same file is still in cache, but I have a hard time to reproduce that and don't even know if it's possible. Bug does still exist in some form anyway.
